Download binaries from `vim`

abitrolly opened the pull request:
`:LanguageClientInstallBinaries` makes it easy to download or refresh `languageclient`. This is my first time writing `vimscript` and I just wrapped provided `install.sh`.

abitrolly committed:
:LanguageClientInstallBinaries to download client from `vim`

abitrolly committed:
Document installation process with `:LanguageClientInstallBinaries`

autozimu commented:
What are the benefits of providing this extra command than relying on plugin manager like plug or dein to do the work automatically? If the user needs to do the download one time occasionally, it should be easy enough to fire off the command in terminal.

abitrolly commented:
@autozimu I don't use plugin managers with Vim 8, because it is able to discover and run plugin itself. The ability to download/refresh binaries is a convenience feature I used to in `vim-go`. Instead of directing users to RTFM in case of missing binaries, it offers a quick way to fix problem with one command.

abitrolly committed:
Fix `vint autoload plugin` errors

autoload/LanguageClient.vim:599:1: Use the abort attribute for functions in autoload (see Google VimScript Style Guide (Functions))
autoload/LanguageClient.vim:605:13: Prefer single quoted strings (see Google VimScript Style Guide (Strings))
